src/common/meta/src/wal/kafka/topic_manager.rs

```diff
@@ -12,40 +12,113 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::collections::HashSet;
 use std::sync::Arc;
+use std::time::Duration;
 
-use crate::error::Result;
+use common_telemetry::debug;
+use rskafka::client::ClientBuilder;
+use rskafka::BackoffConfig;
+use snafu::{ensure, ResultExt};
+
+use crate::error::{
+    BuildClientSnafu, BuildCtrlClientSnafu, CreateKafkaWalTopicSnafu, DecodeJsonSnafu,
+    EncodeJsonSnafu, InvalidNumTopicsSnafu, Result,
+};
 use crate::kv_backend::KvBackendRef;
+use crate::rpc::store::PutRequest;
 use crate::wal::kafka::topic::Topic;
 use crate::wal::kafka::topic_selector::{RoundRobinTopicSelector, SelectorType, TopicSelectorRef};
 use crate::wal::kafka::KafkaConfig;
 
+const CREATE_TOPIC_TIMEOUT: i32 = 5_000; // 5,000 ms.
+const CREATED_TOPICS_KEY: &str = "__created_kafka_wal_topics";
+
 /// Manages topic initialization and selection.
 pub struct TopicManager {
+    config: KafkaConfig,
     topic_pool: Vec<Topic>,
     topic_selector: TopicSelectorRef,
     kv_backend: KvBackendRef,
 }
 
 impl TopicManager {
     /// Creates a new topic manager.
-    pub fn new(config: &KafkaConfig, kv_backend: KvBackendRef) -> Self {
+    pub fn new(config: KafkaConfig, kv_backend: KvBackendRef) -> Self {
         let selector = match config.selector_type {
-            SelectorType::RoundRobin => RoundRobinTopicSelector::new(),
+            SelectorType::RoundRobin => RoundRobinTopicSelector::with_shuffle(),
         };
 
         Self {
+            config,
             topic_pool: Vec::new(),
             topic_selector: Arc::new(selector),
             kv_backend,
         }
     }
 
-    /// Tries to initialize the topic pool.
+    /// Tries to initialize the topic manager.
     /// The initializer first tries to restore persisted topics from the kv backend.
-    /// If not enough topics retrieved, the initializer will try to contact the Kafka cluster and request more topics.
+    /// If not enough topics retrieved, the initializer will try to contact the Kafka cluster and request creating more topics.
     pub async fn try_init(&mut self) -> Result<()> {
```

Comment:
We can not invoke the `try_init` on every meta server. Only the leader of the mete server can do this.

Reply:
The try_init method is renamed to start and it's called in MetaStateHandler::on_become_leader.